pisek/task_jobs/data/manager.py

```diff
@@ -32,45 +32,65 @@ def __init__(self) -> None:
         super().__init__("Processing data")
 
     def _get_jobs(self) -> list[Job]:
+        sample_inputs = self.globs_to_files(
+            self._env.config.subtasks[0].all_globs, TaskPath.static_path(self._env, ".")
+        )
         static_inputs = self.globs_to_files(
             self._env.config.input_globs, TaskPath.static_path(self._env, ".")
         )
 
-        if self._env.config.task_type == TaskType.communication:
-            static_outputs = []
-        else:
-            static_outputs = [path.replace_suffix(".out") for path in static_inputs]
-            for static_inp, static_out in zip(static_inputs, static_outputs):
-                if not static_out.exists():
+        static_outputs: list[TaskPath] = []
+        if self._env.config.task_type != TaskType.communication:
+            for static_inp in static_inputs:
+                static_out = static_inp.replace_suffix(".out")
+                if static_out.exists():
+                    static_outputs.append(static_out)
+                elif static_inp in sample_inputs:
                     raise PipelineItemFailure(
-                        f"Missing matching output '{static_out:p}' for static input '{static_inp:p}'."
+                        f"Missing matching output '{static_out:p}' for sample input '{static_inp:p}'."
                     )
 
         all_static_inputs = self.globs_to_files(
             ["*.in"], TaskPath.static_path(self._env, ".")
         )
-        all_input_infos: list[InputInfo] = [
-            InputInfo.static(inp.name.removesuffix(".in")) for inp in all_static_inputs
+        all_testcase_infos: list[TestcaseInfo] = [
+            (
+                TestcaseInfo.static(inp.name.removesuffix(".in"))
+                if self._env.config.task_type == TaskType.communication
+                or inp.replace_suffix(".out").exists()
+                else TestcaseInfo.mixed(inp.name.removesuffix(".in"))
+            )
+            for inp in all_static_inputs
```

Comment:
This logic is partially duplicated from the loop above. The two loops could be merged.